devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,219 @@
+        run(
+            [
+                "cmake",
+                f"-S {str(self.directory)}/gromacs-repo",
+                f"-B {self.gromacs_build_path}",
+                f"-DCMAKE_BUILD_TYPE=Release",
+                f"-DCMAKE_CXX_COMPILER=clang++",
+                f"-DCMAKE_C_COMPILER=clang",
+                f"-DGMX_GPU=SYCL",
+                f"-DGMX_SYCL_ENABLE_GRAPHS=ON",
+                f"-DGMX_FFT_LIBRARY=MKL",
+                f"-DGMX_BUILD_OWN_FFTW=ON",
+                f"-DGMX_GPU_FFT_LIBRARY=MKL",
+                f"-DGMX_GPU_NB_CLUSTER_SIZE=8",
+                f"-DGMX_OPENMP=OFF",
+            ],
+            add_sycl=True,
+        )
+        run(
+            f"cmake --build {self.gromacs_build_path} -j {options.build_jobs}",
+            add_sycl=True,
+        )
+        download(
+            self.directory,
+            self.grappa_url(),
+            self.directory / self.grappa_file(),
+            checksum="cc02be35ba85c8b044e47d097661dffa8bea57cdb3db8b5da5d01cdbc94fe6c8902652cfe05fb9da7f2af0698be283a2",
+            untar=True,
+        )
+
+    def teardown(self):
+        pass
+
+
+class GromacsBenchmark(Benchmark):
+    def __init__(self, suite, model, type):
+        self.suite = suite
+        self.model = model  # The model name (e.g., "0001.5")
+        self.type = type
+        self.gromacs_src = suite.gromacs_src
+        self.grappa_dir = suite.grappa_dir
+        self.gmx_path = suite.gromacs_build_path / "bin" / "gmx"
+
+        if self.type == "pme":
+            self.extra_args = [
+                "-pme",
+                "gpu",
+                "-pmefft",
+                "gpu",
+                "-notunepme",
+            ]
+        else:
+            self.extra_args = []
+
+    def name(self):
+        return f"gromacs-{self.model}-{self.type}"
+
+    def setup(self):
+        if self.type != "rf" and self.type != "pme":
+            raise ValueError(f"Unknown benchmark type: {self.type}")
+
+        if not self.gmx_path.exists():
+            raise FileNotFoundError(f"gmx executable not found at {self.gmx_path}")
+
+        model_dir = self.grappa_dir / self.model
+
+        if not model_dir.exists():
+            raise FileNotFoundError(f"Model directory not found: {model_dir}")
+
+        cmd_list = [
+            str(self.gmx_path),
+            "grompp",
+            "-f",
+            f"{str(self.grappa_dir)}/{self.type}.mdp",
+            "-c",
+            str(model_dir / "conf.gro"),
+            "-p",
+            str(model_dir / "topol.top"),
+            "-o",
+            f"{str(model_dir)}/{self.type}.tpr",
+        ]
+
+        # Generate configuration files
+        self.conf_result = run(
+            cmd_list,
+            add_sycl=True,
+        )
+
+    def run(self, env_vars):
+        model_dir = self.grappa_dir / self.model
+
+        env_vars.update(
+            {
+                "SYCL_CACHE_PERSISTENT": "1",
+                "GMX_CUDA_GRAPH": "1",
```

Comment:
I'm guessing this controls whether we use eager vs graph?

Can you make this configurable and add "eager"/"graph" variants of the benchmarks?